Addition of metadata scheme and signalling of SAP type and earliest PTS (#3)
wilaw/carp: draft-law-moq-carp.md

Every Track entry in a CARP catalog MUST declare a "packaging" type value of "cmaf".

## Timeline description
This specification extends the METADATA element of the timeline track, defined in WARP
Section 7, in the following four aspects: 

* Specification of a general scheme for metadata signalled through the METADATA element
  of the timeline track [Ed.Note: This aspect should also be applied to WARP, thus
  should be moved to WARP later on.]
* Definition of a namespace for CARP-specific metadata signalled through the METADATA
  element of the timeline track
* Addition of metadata signalling of SAP type
* Addition of metadata signalling of earliest presentation time

### General metadata scheme
When not empty, the string in the METADATA field MUST contain one or more comma
separated key=value pairs, formatted as Strings as specified in Section 3.3.3 of

> [RFC9651]. For example, the METADATA field can be "key1=1,key2=3268".

Reviewer: Should you also include an example in which a string value is presented? For example

"key1=1,key2=""hello-world"",key2=3268".

### Namespace for CARP-specific metadata signalled through the METADATA element
For CARP-specific metadata signalled through the METADATA element, the namespace

> is "ietf:moq:warp:timeline:metadata:carp".

Reviewer: Where is this namespace added? Is it the prefix to the key? If so, would be good to explicitly state that. 

```
ietf:moq:warp:timeline:metadata:carp:SAP_TYPE
ietf:moq:warp:timeline:metadata:carp:EARLIEST_PTS
```

Also, what is the separator between the prefix and the key name. Is it : or . or something else?

```
ietf:moq:warp:timeline:metadata:carp.SAP_TYPE
ietf:moq:warp:timeline:metadata:carp.EARLIEST_PTS
```

> is "ietf:moq:warp:timeline:metadata:carp".

Reviewer: `ietf:moq:warp:timeline:metadata:carp`

Does it need to be this verbose? Given that we are working within the WARp specification and this namespace will never be applied outside of that, we could probably shorten to

`warp:timeline:metadata:carp`

or even to 

`timeline:metadata:carp`

### Metadata signalling of SAP type
When the key of a key=value pair is "CARP.SAP_TYPE", the value indicates the SAP
type the Object begins with. The value 0 indicates that the Object does not start with
an ISOBMFF stream access point. The value equal to 1, 2, or 3 indicates that the Object
begins with a stream access point of SAP type 1, 2, or 3, respectively. When the Object
is the first Object in the Group, the value MUST be equal to 1 or 2.

### Metadata signalling of earliest presentation time

> When the key of a key=value pair is "CARP.EARLIEST_PTS", the value indicates the earlier

Reviewer: ```suggestion
When the key of a key=value pair is "CARP.EARLIEST_PTS", the value indicates the earliest
```

media presentation timestamp of all media samples in the Object. When the SAP type of

> the Object begins with is equal to 2 or 3, this key=value pair SHOULD be present.

Reviewer: ```suggestion
the Object is 2 or 3, the CARP.EARLIEST_PTS key SHOULD be present.
```
